scripts/modules/language-detection.js

```
@@ -0,0 +1,376 @@
+		/(?:^|\s)(\/[^\s]+)(?:\s|$)/g,
+		// Windows paths
+		/(?:^|\s)([A-Za-z]:[\/\\][^\s]+)(?:\s|$)/g,
+		// Common filename patterns without paths
+		/(?:^|\s)((?:dockerfile|makefile|rakefile|gemfile|podfile)(?:\.[a-zA-Z0-9]+)?)(?:\s|$)/gi
+	];
+
+	const filePaths = new Set();
+
+	for (const pattern of patterns) {
+		let match;
+		while ((match = pattern.exec(content)) !== null) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)